Makefile

```diff
@@ -448,25 +448,27 @@ staging: doc/CHANGES
 	# Delete the tmp files
 	cd ${TMPDIR} && /bin/rm -rf daikon daikon.dist old_tar.txt new_tar.txt
 
-# Copy the files in the staging area to the website.  This will copy
-# all of the files in staging, but will not delete any files in the website
-# that are not in staging.
-staging-to-www: $(STAGING_DIR)
-#copy the files
+# Copy the files in the staging area to the website.
+# We do this by deleting the previous release directory and
+# renaming the staging directory to be the release directory.
+staging-to-www: $(STAGING_DIR) save-current-release
+	@echo "staging-to-www is disabled while we test new layout."
+	@exit 1
+	-chmod u+w $(WWW_PARENT)
 	-chmod -R u+w $(WWW_DIR)
-# # remove previous release archive files
-# 	\rm -f /cse/web/research/plse/daikon/download/daikon-*
-# # don't trash existing history directory
-# #	(cd $(STAGING_DIR) && tar cf - --exclude=history .) | (cd $(WWW_DIR) && tar xfBp -)
-# Remove previous files, except history and staging-daikon
-	(cd $(STAGING_DIR) && \ls | grep -v history | grep -v staging-daikon | xargs \rm -rf
-# Copy directory
-	rsync -Cavz --quiet $(STAGING_DIR) $(WWW_DIR)
-#	-chmod -R u-w $(WWW_DIR)
+	-chmod -R u+w $(STAGING_DIR)
+# Move 'permanent' directories that were not included in staging
+	\mv $(WWW_DIR)/history $(STAGING_DIR)/history
+	\mv $(WWW_DIR)/download/cvs $(STAGING_DIR)/download/cvs
+# Remove previous release
+	\rm -rf $(WWW_DIR)
+# Rename staging
+	\mv $(STAGING_DIR) $(WWW_DIR)
 # This command updates the dates and sizes in the various index files
-	html-update-link-dates $(DIST_DIR)/index.html
-# with new version number system, this is now done manually
-#	$(MAKE) update-dist-version-file
+	html-update-link-dates $(WWW_DIR)/download/index.html
```

Comment:
Shouldn't this have been done by "make staging"?  I would think that the staging-to-www job would do nothing but copy files, without changing any contents.  (And, this command shouldn't end up making any changes anyway, should it?)